inst/stan/functions/obs_lpmf.stan

```diff
@@ -1,18 +1,19 @@
-real obs_lpmf(int[] dummy, int start, int end, int[] obs, int[] sl, int[] csl,
+real obs_lpmf(int[] obs, int[,] obs_miss, int dmax, int[] sl, int[] csl, int n_groups,
               vector[] imp_obs, int[] sg, int[] st, int[,] rdlurd,
-              vector srdlh, matrix ref_lh, int[] dpmfs, int ref_p, real phi) {
+              vector srdlh, matrix ref_lh, int[] dpmfs, int ref_p, vector[] alpha, real phi) {
   real tar = 0;
   real tar_obs;
-  int start_n = csl[start] - sl[start];
-  int end_n = csl[end];
-  int n = end_n - start_n;
-  int snap_obs[n] = obs[(start_n + 1):end_n];
-  vector[n] exp_obs;
+  real tar_alpha;
+  int n_snaps = num_elements(st);
+  int n_obs = num_elements(obs);
+  int n_obs_miss = num_elements(obs_miss[1]);
+  vector[n_obs] exp_obs;
+  vector[n_obs_miss] exp_obs_miss[n_groups] = rep_array(rep_vector(0, n_obs_miss), n_groups);
   int g, t, l;
   int ssnap = 1;
   int esnap = 0;
 
-  for (i in start:end) {
+  for (i in 1:n_snaps) {
```

Comment:
The reduce sum functionality has been removed and I simplified some elements (e.g. start_n and end_n turned out to be senseless.

Reply:
We probably want to make this optional before going into a release candidate but good for now.

Reply:
After having a chat we agreed I am going to try and implement dual likelihood functions with the missingness model in parallel across groups.